src/lib/math/pcurves/pcurves.h

```diff
@@ -310,18 +310,14 @@ class PrimeOrderCurve {
       *
       * This is currently only supported for a few specific curves
       */
-      virtual AffinePoint hash_to_curve_nu(std::string_view hash,
-                                           std::span<const uint8_t> input,
-                                           std::span<const uint8_t> domain_sep) const = 0;
+      virtual AffinePoint hash_to_curve_nu(std::function<void(std::span<uint8_t>)> expand_message) const = 0;
```

Comment:
Even though this is an internal interface, I think the parameter requires a bit of documentation, e.g.:

`@param expand_message  a callback acting as a XOF generating a uniform stream of bytes`

Reply:
Done